### PR TITLE
Add GSL library to LDFLAGS, use alibuild-generate-module for module g…

### PR DESCRIPTION
**Pull request by mfasDa**
…eneration

### CODE REVIEW

**Review by preghenella — DISMISSED** [review]

**Comment by mfasDa** [comment]
Adding -L${GSL_ROOT}/lib to LD_LIBRARY_PATH is done to overcome an issue found in the creation of the daily builds:
```
2020-10-12@17:18:31:DEBUG:AliGenerators:Herwig:34923a30: /jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156182/1/slc7_x86-64/GCC-Toolchain/v7.3.0-alice2-11/bin/../lib/gcc/x86_64-unknown-linux-gnu/7.3.0/../../../../x86_64-unknown-linux-gnu/bin/ld: cannot find -lgsl
2020-10-12@17:18:31:DEBUG:AliGenerators:Herwig:34923a30: /jenkins/workspace/DailyBuilds/DailyAliGenerators/sw/20156182/1/slc7_x86-64/GCC-Toolchain/v7.3.0-alice2-11/bin/../lib/gcc/x86_64-unknown-linux-gnu/7.3.0/../../../../x86_64-unknown-linux-gnu/bin/ld: cannot find -lgslcblas
```
However it is not clear to me what is the reason for this, as we already specify the GSL location when configuring Herwig.

The move to alibuild-generate-module is done for Herwig, OpenLoops and SHERPA in order to let the tool handle the 
dependencies instead of them being handled by hand.

File: herwig.sh [hunk 1]
```diff
@@ -31,27 +32,15 @@ export LDFLAGS="-L$LHAPDF_ROOT/lib -L$CGAL_ROOT/lib -L$GMP_ROOT/lib $LDZLIB"
...
+mkdir -p etc/modulefiles
+alibuild-generate-module > etc/modulefiles/$PKGNAME
+cat >> etc/modulefiles/$PKGNAME <<EoF
 # Our environment
 set HERWIG_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
 setenv HERWIG_ROOT \$HERWIG_ROOT
 setenv HERWIG_INSTALL_PATH \$::env(HERWIG_ROOT)/lib/Herwig
 prepend-path PATH \$HERWIG_ROOT/bin
-prepend-path LD_LIBRARY_PATH \$HERWIG_ROOT/lib/Herwig
+prepend-path LD_LIBRARY_PATH \$HERWIG_ROOT/lib/Herwigb
```
Comment (maireiphc) [inline herwig.sh:44]: Maybe my ignorance of alibuild-recipe-tools.sh, but ... is this "b" at the end of the line "../lib/Herwigb" added on purpose ?
Same is done apparently for OpenLoops.sh

Comment (mfasDa) [reply herwig.sh:44]: @maireiphc Thanks for spotting! As you can imagine this is not a feature of alibuild-recipe-tools but a simple typo, not sure how it entered. I will fix immediately!

**Review by preghenella — APPROVED** [review]

**Comment by ktf** [comment]
@MichaelLettrich any idea why the macOS build died?

**Comment by preghenella** [comment]
@MichaelLettrich `macOS` failure does not seem to be related to this PR
@ktf can you merge it? Thanks

**Comment by MichaelLettrich** [comment]
@preghenella I know that everyone wants their PRs to be merged as fast as possible and it's frustrating to be stuck waiting for others, but there is no way to ensure that your software is going to build fine using the Apple Xcode compilers and won't cause a similar issue as the Monitoring package does currently. So I kindly ask for patience until things are working again.

**Comment by ktf** [comment]
@MichaelLettrich I think this is not tested in any case by the macOS builder, only by AliGenerators/generators. Merging.

**Comment by mfasDa** [comment]
Thanks! The problem is that the build error mentioned above blocks deployment of aligenerators, already since one week, and within aligenmc which is needed for the setup of a production https://alice.its.cern.ch/jira/browse/ALIROOT-8558, and aligenmc is still under development so currently frequent tags are needed. As we don't have access to the machine which is doing the cvmfs builds it is hard for us to debug which is the actual cause of the problem. With the pull request we basically **try** to cure it, but no way to reproduce unless we have the same setup. Since this is needed for the production setup we need to fix the aligenerators deployment as soon as possible in order to not wait another week in order to continue.

**Comment by mfasDa** [comment]
Just checked myself to build on Xcode12, I see a problem in Rivet:
```
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: cd analists && \
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: 	LD_LIBRARY_PATH=/Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/src/.libs:/Users/markusfasel/alice/sw/osx_x86-64/HepMC/HEPMC_02_06_10-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/fastjet/v3.3.3_1.042-alice1-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/cgal/4.12.2-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/MPFR/v3.1.3-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/GMP/v6.0.0-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/YODA/v1.8.2-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/boost/v1.72.0-alice2-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/defaults-release/v1-1/lib \
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:     DYLD_LIBRARY_PATH=/Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/src/.libs: \
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:     PYTHONPATH=/Users/markusfasel/alice/sw/osx_x86-64/YODA/v1.8.2-1/lib/python2.7/site-packages:/Users/markusfasel/alice/sw/INSTALLROOT/2271a694c25f10cce2568f5413b10d816a034b44/osx_x86-64/Rivet/3.1.1-alice1-1/lib/python2.7/site-packages:/Users/markusfasel/alice/sw/osx_x86-64/YODA/v1.8.2-1/lib64/python2.7/site-packages:/Users/markusfasel/alice/sw/osx_x86-64/YODA/v1.8.2-1/lib/python2.7/site-packages \
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:     RIVET_ANALYSIS_PATH=/Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/analyses:: \
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:     /usr/bin/python /Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/doc/analists/mk-analysis-txt || true
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: Using output file name 'analyses.dat'
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: Traceback (most recent call last):
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:   File "/Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/doc/analists/mk-analysis-txt", line 18, in <module>
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:     import rivet
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:   File "/Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/pyext/build/lib.macosx-10.15-x86_64-2.7/rivet/__init__.py", line 12, in <module>
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:     from rivet.core import *
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: ImportError: dlopen(/Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/pyext/build/lib.macosx-10.15-x86_64-2.7/rivet/core.so, 10): Symbol not found: ___gmpn_add_n
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:   Referenced from: /Users/markusfasel/alice/sw/osx_x86-64/fastjet/v3.3.3_1.042-alice1-1/lib/libfastjet.0.dylib
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:   Expected in: flat namespace
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:  in /Users/markusfasel/alice/sw/osx_x86-64/fastjet/v3.3.3_1.042-alice1-1/lib/libfastjet.0.dylib
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: cd analists && \
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: 	LD_LIBRARY_PATH=/Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/src/.libs:/Users/markusfasel/alice/sw/osx_x86-64/HepMC/HEPMC_02_06_10-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/fastjet/v3.3.3_1.042-alice1-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/cgal/4.12.2-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/MPFR/v3.1.3-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/GMP/v6.0.0-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/YODA/v1.8.2-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/boost/v1.72.0-alice2-1/lib:/Users/markusfasel/alice/sw/osx_x86-64/defaults-release/v1-1/lib \
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:     DYLD_LIBRARY_PATH=/Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/src/.libs: \
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:     PYTHONPATH=/Users/markusfasel/alice/sw/osx_x86-64/YODA/v1.8.2-1/lib/python2.7/site-packages:/Users/markusfasel/alice/sw/INSTALLROOT/2271a694c25f10cce2568f5413b10d816a034b44/osx_x86-64/Rivet/3.1.1-alice1-1/lib/python2.7/site-packages:/Users/markusfasel/alice/sw/osx_x86-64/YODA/v1.8.2-1/lib64/python2.7/site-packages:/Users/markusfasel/alice/sw/osx_x86-64/YODA/v1.8.2-1/lib/python2.7/site-packages \
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:     RIVET_ANALYSIS_PATH=/Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/analyses:: \
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:     /usr/bin/python /Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/doc/analists/mk-analysis-json || true
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: Using output file name 'analyses.json'
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: Traceback (most recent call last):
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:   File "/Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/doc/analists/mk-analysis-json", line 18, in <module>
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:     import rivet
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:   File "/Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/pyext/build/lib.macosx-10.15-x86_64-2.7/rivet/__init__.py", line 12, in <module>
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:     from rivet.core import *
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: ImportError: dlopen(/Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/pyext/build/lib.macosx-10.15-x86_64-2.7/rivet/core.so, 10): Symbol not found: ___gmpn_add_n
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:   Referenced from: /Users/markusfasel/alice/sw/osx_x86-64/fastjet/v3.3.3_1.042-alice1-1/lib/libfastjet.0.dylib
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:   Expected in: flat namespace
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2:  in /Users/markusfasel/alice/sw/osx_x86-64/fastjet/v3.3.3_1.042-alice1-1/lib/libfastjet.0.dylib
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: /usr/local/bin/gmkdir -p /Users/markusfasel/alice/sw/INSTALLROOT/2271a694c25f10cce2568f5413b10d816a034b44/osx_x86-64/Rivet/3.1.1-alice1-1/share/Rivet
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: /bin/sh /Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/install-sh -c -m 644 doc/analists/analyses.dat /Users/markusfasel/alice/sw/INSTALLROOT/2271a694c25f10cce2568f5413b10d816a034b44/osx_x86-64/Rivet/3.1.1-alice1-1/share/Rivet/
2020-10-15@12:07:57:DEBUG:Herwig:Rivet:aliceo2: /Users/markusfasel/alice/sw/BUILD/2271a694c25f10cce2568f5413b10d816a034b44/Rivet/install-sh: doc/analists/analyses.dat does not exist.
```

**Pull request merged**